sky/resources.py

```diff
@@ -648,6 +704,20 @@ def _get_default_runtime_version() -> str:
         self._accelerators = accelerators
         self._accelerator_args = accelerator_args
 
+    def _set_autostop_config(
+        self,
+        autostop: Union[bool, int, Dict[str, Any], None],
+    ) -> None:
+        # Unset, use default (usually disabled, but enabled for controllers.)
+        self._autostop_config: Optional[AutostopConfig] = None
+        # Otherwise, convert based on the type of config schema.
+        if isinstance(autostop, bool):
+            self._autostop_config = AutostopConfig.from_bool(autostop)
+        elif isinstance(autostop, int):
+            self._autostop_config = AutostopConfig.from_int(autostop)
+        elif isinstance(autostop, dict):
+            self._autostop_config = AutostopConfig.from_yaml_config(autostop)
```

Comment:
very minor: it is generally a better idea to keep the API of a class to take ambiguous inputs. We can use a method `AutostopConfig.from_val(autostop)` and decide the type in that method.

Reply:
refactored